### PR TITLE
feat: track dial failures

### PR DESCRIPTION
**Pull request by adklempner**

### CODE REVIEW

File: lib/metrics/dial_failure.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	"github.com/status-im/telemetry/pkg/types"
+)
+
+type DialFailure struct {
+	GenericMetric[types.DialFailure]
+}
+
+var (
+	DialFailureProcessor = &DialFailure{}
+)
```
Comment (vpavlin) [inline lib/metrics/dial_failure.go:13]: This is missing the implementation isn't it?

Comment (adklempner) [reply lib/metrics/dial_failure.go:13]: Nope! After my generic metrics PR this is all that is necessary to add all the db logic 😄 https://github.com/status-im/telemetry/pull/56/files#diff-c4d76f1754d2a1b8e04e65bde83777fa2509c22740f0dff2d8e8a7efcb19ae3b

Comment (adklempner) [reply lib/metrics/dial_failure.go:13]: All the column names and values necessary to create the insert query are derived based on the json attributes of the `DialFailure` type. As long as the name of the field matches the name of the column, it works. https://github.com/status-im/telemetry/blob/53fa8accec5d37e2899314a160702a5909eb2948/lib/metrics/generic.go#L103

Comment (vpavlin) [reply lib/metrics/dial_failure.go:13]: Oh, right, I completely forgot. That's awesome!

File: lib/database/sql/000020_dial_failure.up.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    id SERIAL PRIMARY KEY,
+    recordId INTEGER NOT NULL,
+    errorType INTEGER NOT NULL,
+    errorMsg TEXT,
+    protocols TEXT NOT NULL,
+    timestamp INTEGER NOT NULL,
+    CONSTRAINT dialFailure_unique UNIQUE (recordId, errorType, protocols, timestamp)
+);
+
+ALTER TABLE dialFailure ADD CONSTRAINT fk_dialFailure_telemetryRecord
```
Comment (vpavlin) [inline lib/database/sql/000020_dial_failure.up.sql:11]: Shouldn't there be a foreign key constraint for `dial_error_types` as well? Just to make sure we don't insert non-existant error types?

**Review by vpavlin — APPROVED** [review]
Added one more comment, but otherwise looks good

**Pull request merged**